src/apis/stockx/asks/index.js

```diff
@@ -11,121 +12,142 @@ export default class Asks extends Base {
-        const error = new Error('Invalid product, amount, and/or size!');
-        error.status = 404;
-        throw error;
-      }
-
-      if (!bearer) {
-        const error = new Error('Please login first!');
-        error.status = 401;
-        throw error;
-      }
-
-      const desiredSize = /random/i.test(size) ? randomInclusive(product.variants) : product.variants.find(v => v.size === size);
-
-      if (!desiredSize || (desiredSize && !desiredSize.uuid)) {
-        const error = new Error('Size not found!');
-        error.status = 404;
-        throw error;
-      }
-
-      const { uuid } = desiredSize;
-
-      const res = await request('https://stockx.com/api/portfolio?a=ask', {
-        headers: {
-          ...headers,
-          authorization: `Bearer ${bearer}`,
-          'content-type': 'application/json',
-        },
-        jar,
-        json: {
-          PortfolioItem: {
-            localAmount: amount,
-            skuUuid: uuid,
-            localCurrency: currency,
-            expiresAt
-          },
-        },
-        method: 'POST',
-        proxy,
-        resolveWithFullResponse: true,
-        simple: false,
-      });
-
-      const { statusCode, body } = res;
-      if (!statusCode || (statusCode && statusCode !== 200)) {
-        const err = new Error('Invalid response code!');
-        err.status = statusCode || 404;
-        throw err;
-      }
-
-      const { PortfolioItem: { chainId, skuUuid }} = body;
-      return { chainId, skuUuid };
-    } catch (error) {
-      return errors(error, 'place ask');
+    if (!amount || !size || !product) {
+      const error = new Error('Invalid product, amount, and/or size!');
+      error.status = 404;
+      throw error;
+    }
+
+    if (!bearer) {
+      const error = new Error('Please login first!');
+      error.status = 401;
+      throw error;
     }
+
+    const desiredSize = /random/i.test(size) ? randomInclusive(product.variants) : product.variants.find(v => v.size === size);
+
+    if (!desiredSize || (desiredSize && !desiredSize.uuid)) {
+      const error = new Error('Size not found!');
+      error.status = 404;
+      throw error;
+    }
+
+    const { uuid } = desiredSize;
+
+    const res = await request('https://stockx.com/api/portfolio?a=ask', {
+      headers: {
+        ...headers,
+        authorization: `Bearer ${bearer}`,
+        'content-type': 'application/json',
+      },
+      jar,
+      json: {
+        PortfolioItem: {
+          localAmount: amount,
+          skuUuid: uuid,
+          localCurrency: currency,
+          expiresAt
+        },
+      },
+      method: 'POST',
+      proxy,
+      resolveWithFullResponse: true,
+      simple: false,
+    });
+
+    checkStatus(res);
+
+    const { body } = res;
+    const { PortfolioItem: { chainId, skuUuid }} = body;
+
+    return { chainId, skuUuid };
   }
 
-  // TODO!
-  async remove(options = {}) {}
+  async remove(bid = {}) {
+    const askID = bid.chainId;
```

Comment:
We should probably keep it consistent here and just use destructuring.
```suggestion
    const { chainId } = bid;
```